src/extension/utils/createAndOpenConfFile.ts

```diff
@@ -90,14 +128,268 @@ function convertSourceFormDataToConfFileJSON(
   if (inputFormData.additionalSettings?.length) {
     inputFormData.additionalSettings.forEach(({ option, value }) => {
       if (option) {
-        config[option] = setAdditionalSetting(value)
+        config[option] = setAdditionalSetting(value as string)
       }
     })
   }
 
   return JSON.stringify(config, null, 2)
 }
 
+/**
+ * translate 'link' attribute between forms
+ * @param form InputFormData
+ * @param solObj SilidityObj
+ * @param contractName name of the contract that has the link
+ */
+function processLink(
+  form: InputFormData,
+  solObj: SolidityObj,
+  contractName: string,
+) {
+  const linking = solObj.linking
+  if (linking.length > 0) {
+    linking.forEach(link => {
+      if (link.variable && link.contractName) {
+        form.link.push({
+          id: '',
+          contractName: contractName,
+          fieldName: link.variable,
+          associatedContractName: link.contractName,
+        })
+      }
+    })
+  }
+}
+
+/**
+ * add additional setting with flag [flag] and value [value] the the form
+ * @param flag flag (string)
+ * @param value value (string)
+ * @param form to add additional setting to
+ */
+function addAdditionalSetting(
+  flag: string,
+  value: string,
+  form: InputFormData,
+) {
+  if (value) {
+    form.additionalSettings.push({
+      id: flag,
+      option: flag,
+      value: value,
+    })
+  }
+}
+
+function addSolcArguments(
+  flag: string,
+  solcArgs: SolcArg[],
+  form: InputFormData,
+) {
+  if (solcArgs) {
+    let strSolcArgs = '['
+    solcArgs.forEach(arg => {
+      if (arg.key && arg.value) {
+        strSolcArgs += '--' + arg.key + ',' + arg.value + ','
+      } else if (arg.key) {
+        strSolcArgs += '--' + arg.key + ','
+      }
+    })
+    if (strSolcArgs !== '[') {
+      strSolcArgs = strSolcArgs.replace(/.$/, ']')
+      form.additionalSettings.push({
+        id: flag,
+        option: flag,
+        value: strSolcArgs,
+      })
+    }
+  }
+}
+
+/**
+ * add additional contracts information to the [form]
+ * @param solidityAdditionalContracts additional contracts to add
+ * @param form to add contracts to
+ */
+function processAdditionalContracts(
+  solidityAdditionalContracts: SolidityObj[],
+  form: InputFormData,
+) {
+  form.useAdditionalContracts = true
+  form.solc_map.push({
+    contract: form.mainContractName,
+    solidityCompiler: form.solidityCompiler,
+  })
+
+  solidityAdditionalContracts.forEach(solObj => {
+    if (solObj.mainContract) {
+      form.additionalContracts.push({
+        file: solObj.mainFile.value?.toString(),
+        name: solObj.mainContract,
+      })
+
+      processLink(form, solObj, solObj.mainContract)
+
+      if (solObj.compiler.ver) {
+        let compDir: string = solObj.compiler.exe || ''
+        if (compDir !== '') {
+          compDir += '/'
```

Comment:
make sure this is compatible with windows / other platforms

Reply:
added a check to see if the path given in compDir is valid, using vscode.Uri object